File: album_art.py
# After hours of trying,
# I couldn't get the ruby script working
# So, I made a similar one for python

import sys
import pylast
import yaml
import MySQLdb
import os

API_KEY = "1046bf7d632bb797c7d3430962cc2549"
API_SECRET = "a973b39e259ec530d08d0daad7ac718d"

network = pylast.get_lastfm_network(API_KEY, API_SECRET)

print "Initialized all modules"

dbconfig = yaml.load(open('database.yml', 'r'))
db = MySQLdb.connect(dbconfig['host'], dbconfig['username'], dbconfig['password'], dbconfig['database'])

print "Initialized database"

try:
    pics_folder = sys.argv[1]
except Exception:
    raise SystemExit

cursor = db.cursor()
cursor.execute('SELECT * FROM albums WHERE language=\'English\'')
result = cursor.fetchall()
errors = []

for row in result:
    try:
        album_name = row[1]
        album_id = str(row[0])
        filename = pics_folder + '/' + album_id + '.jpg'
        if os.path.exists(filename):
            continue
        cursor.execute('SELECT artist FROM tracks WHERE album=\'' + album_id + '\'')
        artist_name = cursor.fetchall()[0][0]
        album_obj = pylast.Album(artist_name, album_name, network)
        album_uri = album_obj.get_cover_image()
        bash_command = 'wget -nv ' + album_uri + ' -O ' + filename
        print album_name, artist_name
        os.system(bash_command)
    except Exception:
        errors.append([album_name, artist_name])

print errors

File: album_art.rb
#!/usr/bin/env ruby
require 'rubygems'	#Gem Support
require 'lastfm'
require 'pathname'
require 'active_record'#ORM
require 'logger'	#database.log
require 'funcs'
require 'models'	#Track,Album etc
require 'yaml'
require 'json'

##Setup the database

puts "Initialized all gems"

dbconfig = YAML::load(File.open('database.yml'))
ActiveRecord::Base.establish_connection(dbconfig)
ActiveRecord::Base.logger = Logger.new(File.open('database.log', 'a'))
ActiveSupport::LogSubscriber.colorize_logging=false
ActiveRecord::Base.logger.formatter = proc { |severity, datetime, progname, msg|
  "#{msg}\n"
}
puts "initialized database"

pics_folder=ARGV[0]
exit if pics_folder.length==0
lastfm = Lastfm.new("1046bf7d632bb797c7d3430962cc2549", "a973b39e259ec530d08d0daad7ac718d")
Album.all(:conditions => { :language => 'English' }, :order=>"id DESC").each do |album|
	puts album.name
	next if File.exist?("#{pics_folder}/#{album.id}.jpg")
	bandName = Band.find(Track.find_by(:album_id, album.id).band).name
	begin
		album_info = lastfm.album.get_info(bandName,album.name)
	rescue
		next
	end
	next unless album_info['image'].length >0
	album_info['image'].each do |img|
		if img["size"]=='large' and img['content']
			puts img["content"]
			filename = "#{pics_folder}/#{album.id}.jpg"
			`wget -nv "#{img["content"]}" -O #{filename}`
		end
	end
end
